### PR TITLE
AG-13233 - Add styleContainer to avoid detached DOM causing styling issues.

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-13233

Published as `10.3.0-beta.20241022.1441`

### CODE REVIEW

File: packages/ag-charts-community/src/chart/chartProperties.ts [hunk 1]
```diff
@@ -25,6 +25,9 @@ export class ChartProperties extends BaseProperties {
     @Validate(OBJECT.restrict(HTMLElement), { optional: true })
     container?: HTMLElement;
 
+    @Validate(OBJECT.restrict(HTMLElement), { optional: true })
```
Comment (iMoses) [inline packages/ag-charts-community/src/chart/chartProperties.ts]: this file is actually dead code 😶

File: packages/ag-charts-community/src/chart/chart.ts [hunk 2]
```diff
@@ -246,6 +247,7 @@ export abstract class Chart extends Observable {
 
         const scene: Scene | undefined = resources?.scene;
         const container = resources?.container ?? options.processedOptions.container ?? undefined;
+        const styleContainer = resources?.styleContainer ?? options.processedOptions.styleContainer ?? undefined;
```
Comment (iMoses) [inline packages/ag-charts-community/src/chart/chart.ts]:
```suggestion
        const styleContainer = resources?.styleContainer ?? options.processedOptions.styleContainer;
```
nit: `?? undefined` is redundant.

File: packages/ag-charts-community/src/chart/chartProperties.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/chartProperties.ts]: Removed :+1:

File: packages/ag-charts-community/src/chart/chart.ts (hunk 2 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/chart.ts]: Fixed

**Pull request merged**